File: source/downloads/code/LeetCode/Java/Find-Minimum-in-Rotated-Array.java
public class Solution {
    public int findMin(int[] nums) {
        int l = 0;
        int r = nums.length-1;
        if (nums.length==1)
	    return nums[0];
        while (l<r){
            int m = l+(r-l)/2;
            if (m > 0 && nums[m]<nums[m-1])
                return nums[m];
            if (nums[l]<=nums[m] && nums[m] > nums[r])
                l = m+1;
            else
                r = m-1;;
        }
        return nums[l];        
    }
}

File: downloads/code/LintCode/Find-Minimum-in-Rotated-Array.py
class Solution:
    # @param num: a rotated sorted array
    # @return: the minimum number in the array
    def findMin(self, num):
        # write your code here
        if num == None or len(num) == 0:
            return 0
        front = 0
        end = len(num)-1

        result = sys.maxint
        while front <= end and front >=0:
            mid = (front+end)/2
            # case 1: no in orded subarray
            if num[front] > num[end] :
                front = mid+1
            else:
                # case 2: already in a orded subarray, search for the lowest value
                if result > num[front]:
                    result = num[front]
                front -= 1
                end = mid - 1

        return result
